aspnetcore/fundamentals/logging/loggermessage.md

```diff
@@ -0,0 +1,173 @@
+---
+title: High-performance logging with LoggerMessage in ASP.NET Core
+author: guardrex
+description: Learn how to use LoggerMessage features to create cacheable delegates that require fewer object allocations than logger extension methods for high-performance logging scenarios.
+ms.author: riande
+manager: wpickett
+ms.date: 10/31/2017
+ms.topic: article
+ms.assetid: 1725ffe9-b6be-4fbb-af35-c8b9363e2292
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/logging/loggermessage
+---
+# High-performance logging with LoggerMessage in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[LoggerMessage](/dotnet/api/microsoft.extensions.logging.loggermessage) features create cacheable delegates that require fewer object allocations than [logger extension methods](/dotnet/api/Microsoft.Extensions.Logging.LoggerExtensions), such as `LogInformation`, `LogDebug`, and `LogError`. For high-performance logging scenarios, use the `LoggerMessage` pattern.
+
+Logger extension methods require "boxing" (converting) value types, such as `int`, into `object` types and back. The `LoggerMessage` pattern avoids boxing by using static `Action` fields to define logging messages for strongly-typed extension methods. 
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/logging/loggermessage/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The sample app demonstrates `LoggerMessage` features with a simple quote tracking database. The user adds, deletes, and deletes all quotes from an in-memory database. As these operations occur, log messages are generated using static extension methods that trigger logging `Action`s.
+
+## LoggerMessage.Define
+
+[Define(LogLevel, EventId, String)](/dotnet/api/microsoft.extensions.logging.loggermessage.define) creates an `Action` delegate for logging a message. `Define` overloads permit passing up to six type parameters to a named format string (template).
+
+The string provided to the `Define` method is a template and not an interpolated string. Placeholders are filled in the order that the types are specified. Placeholder names in the template should be descriptive and consistent across templates because they serve as property names within structured log data. The sample app uses [Pascal casing](/dotnet/standard/design-guidelines/capitalization-conventions) for placeholder names (for example, "{Count}", "{FirstName}").
```

Comment:
I'd go as far as to say we recommend using Pascal Casing. I think that's what our framework logging uses (ping @BrennanConroy), and it's good for traces to have some consistency there.

Reply:
Turns out we aren't consistent, but we should be. It looks like PascalCase is the most common though.